### ISSUE TITLE
*Feature Request*: Customizable styles on echo area

### ISSUE THREAD

**Issue by gravitational-dark-light** [atlas-engineer/next#582]
For some strange reason %echo-status imposes default styles in the echo area, overriding the ones from *minibuffer-class* 

Wouldn't be more desirable for those to be configurable?

**Comment by Ambrevar** [comment]
I'd like to revamp the echo area into a first class status bar once the
`ffi` branch is merged.  This should resolve a ton of issues including
this one :)

**Comment by Ambrevar** [comment]
#665 implements this.

**Comment by Ambrevar** [comment]
Closed with ac6eb9c94a1612957af722c65d64958e64862b09.

**Comment by Ambrevar** [comment]
@gravitational-dark-light I've added a code example in commit
ce25dea0200e2c4186d537967f9c0c56f7c1fbc9 to do what you want.

**Comment by gravitational-dark-light** [comment]
Thanks, I've already styled the status buffer and the message buffer, and tweaked the formatter to suit my tastes better

I have a question tough: would it be possible to style the text on the status buffer? I'm thinking of something like emacs, with different colors and font weight for both the echo area and modeline

![init-screenshot](https://user-images.githubusercontent.com/8922976/79924321-83d99880-8405-11ea-9fa2-564f2a432b5f.png)

**Comment by Ambrevar** [comment]
The status view is a web view, and as such its content is HTML! :)

You can use `(markup:markup (:b "bold text"))` to write in bold.
For colors, use cl-css.

**Comment by jmercouris** [comment]
Please note that we do not guarantee that the status view will be HTML, it just happens that in the current GTK renderer we have implemented it as a web view

The only guarantee you have is that you can post text to the status and/or message view

How you style them, and what views they are is subject to the renderer's implementation

**Comment by gravitational-dark-light** [comment]
> The status view is a web view, and as such its content is HTML! :) You can use `(markup:markup (:b "bold text"))` to write in bold. For colors, use cl-css.

I tried that today, and somehow, the string is displayed raw, such that I end with `<b>bold text</b>` in the status buffer

**Comment by Ambrevar** [comment]
Fixed with f35770d7fc9fec2002c7b0e388e4113d35044ec3.

I've set my modes in bold with

```
(defun my-format-status (window)
  (declare (ignore window))
  (let* ((buffer (current-buffer))
         (buffer-count (1+ (or (position buffer
                                         (sort (alexandria:hash-table-values (buffers *browser*))
                                               #'<
                                               :key #'id))
                               0))))
    (str:concat
     (markup:markup
      (:b (format nil "[~{~a~^ ~}]"
                  (mapcar (lambda (m) (str:replace-all "-mode" ""
                                                       (str:downcase
                                                        (class-name (class-of m)))))
                          (modes buffer)))))
     (format nil " (~a/~a) ~a — ~a"
             buffer-count
             (hash-table-count (buffers *browser*))
             (url buffer)
             (title buffer)))))

(define-configuration window
  ((status-formatter #'my-format-status)))
```
